File: Multithreading/Future & Promises/C++/Future and Promises.cpp
#include <iostream>
#include <future>
#include <chrono>
#include <thread>

using namespace std;

int async_task() {
    cout << "task running..." << endl;
    this_thread::sleep_for(chrono::seconds(2));
    return 10;
}

int main() {

    future<int> result = async(launch::async, async_task);

    int data = result.get();
    cout<< "result: " << data << endl;

    return 0;
}

File: Multithreading/Future & Promises/Python/future_and_promises.py
from concurrent.futures.thread import ThreadPoolExecutor

def task():
    sum = 0
    for i in range(10):
        sum += i
    return sum

with ThreadPoolExecutor(max_workers=1) as executor:
    future = executor.submit(task)
    print("Waiting for task to be completed")
    print("result: ", future.result())
